QUIC: Run multistream tests in blocking mode as well

hlandau opened the pull request:
This expands the multistream test to run everything in both blocking and non-blocking mode using the same set of scripts. This should dramatically increase our test coverage for our APIs in blocking mode.

Internally this works by spinning up an extra thread to tick the TSERVER so that we can enter a blocking call in the main thread.

Built on the WAIT_PEER PR.

t8m commented:
CI still relevant

hlandau commented:
Yeah, I'm investigating.

hlandau commented:
This is due to our Windows code not having an implementation of `ossl_crypto_condvar_broadcast`. I'm working on a fix.

trixieboyle commented:
> This expands the multistream test to run everything in both blocking and non-blocking mode using the same set of scripts. This should dramatically increase our test coverage for our APIs in blocking mode.
> 
> Internally this works by spinning up an extra thread to tick the TSERVER so that we can enter a blocking call in the main thread.
> 
> Built on the WAIT_PEER PR.

Tammy Boyle

hlandau commented:
Updated and rebased, CI passing, ready for review.

This adds a new Win32 pre-Vista condition variable emulation algorithm, and tests on Windows are now passing.

mattcaswell commented:
Needs another rebase

hlandau commented:
Rebased

mattcaswell commented:
Hmmm....is CI failure relevant?

hlandau commented:
> Hmmm....is CI failure relevant?

It doesn't look like it.

Updated.

hlandau commented:
Merged to master. Thanks for the reviews.
